### ISSUE TITLE
The program doesn't seem to be able to read all the E621 files

### ISSUE THREAD

**Issue by r2ten-catuncle** [McSib/e621_downloader#55]
The search results of krystal in E621 show that there are more than 10,000 image files, but the program only captures 1280 files

[e621_downloader.log](https://github.com/McSib/e621_downloader/files/9005715/e621_downloader.log)
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:56] Printing system information out into log for debug purposes...
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:57] ARCH:           "x86_64"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:58] DLL_EXTENSION:  "dll"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:59] DLL_PREFIX:     ""
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:60] DLL_SUFFIX:     ".dll"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:61] EXE_EXTENSION:  "exe"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:62] EXE_SUFFIX:     ".exe"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:63] FAMILY:         "windows"
22:10:22 [TRACE] (1) e621_downloader: [src\main.rs:64] OS:             "windows"
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:34] Starting e621 downloader...
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:35] Program Name: e621_downloader
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:36] Program Version: 1.7.1
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:37] Program Authors: McSib <mcsibsart@gmail.com>
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:38] Program Working Directory: D:\e621
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:47] Checking if config file exists...
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:55] Checking if tag file exists...
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:69] Login information loaded...
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:70] Login Username: catuncle
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:71] Login API Key: ************************
22:10:22 [TRACE] (1) e621_downloader::program: [src\program.rs:72] Login Download Favorites: true
22:10:22 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:72] SenderClient initializing with USER_AGENT_VALUE "e621_downloader/1.7.1 (by McSib <mcsibsart@gmail.com> on e621)"
22:10:22 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:64] Prompt for safe mode...
22:10:24 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:77] Safe mode decision: true
22:10:24 [TRACE] (1) e621_downloader::program: [src\program.rs:79] Parsing tag file...
22:10:25 [TRACE] (1) e621_downloader::program: [src\program.rs:84] Parsing user blacklist...
22:10:26 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:103] Grabbing posts...
22:10:26 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 1 of tag fav:catuncle
22:10:27 [INFO] [38;5;39m[3m"fav:catuncle"[0m grabbed!
22:10:27 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 1 of tag krystal
22:10:30 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 2 of tag krystal
22:10:33 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 3 of tag krystal
22:10:35 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 4 of tag krystal
22:10:38 [INFO] [38;5;39m[3m"krystal"[0m grabbed!
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:244] Total file size for all images grabbed is 2918854680KB
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "Single Posts"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        ""
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] Collection Post Length:     "0"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/Single Posts"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:217] Collection Single Posts is finished downloading...
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "fav:catuncle"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        ""
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] Collection Post Length:     "0"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/fav_catuncle"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:217] Collection fav:catuncle is finished downloading...
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "krystal"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        "General Searches"
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] **Collection Post Length:     "1280"**
22:10:38 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/General Searches\krystal"

On the E621, the search result of krystal score:>50 shows that there are more than 4500 files
And the files captured by the program are only 377 files

[e621_downloader.log](https://github.com/McSib/e621_downloader/files/9005754/e621_downloader.log)
[TRACE] (1) e621_downloader: [src\main.rs:56] Printing system information out into log for debug purposes...
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:57] ARCH:           "x86_64"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:58] DLL_EXTENSION:  "dll"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:59] DLL_PREFIX:     ""
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:60] DLL_SUFFIX:     ".dll"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:61] EXE_EXTENSION:  "exe"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:62] EXE_SUFFIX:     ".exe"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:63] FAMILY:         "windows"
22:18:51 [TRACE] (1) e621_downloader: [src\main.rs:64] OS:             "windows"
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:34] Starting e621 downloader...
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:35] Program Name: e621_downloader
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:36] Program Version: 1.7.1
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:37] Program Authors: McSib <mcsibsart@gmail.com>
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:38] Program Working Directory: D:\e621
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:47] Checking if config file exists...
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:55] Checking if tag file exists...
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:69] Login information loaded...
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:70] Login Username: catuncle
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:71] Login API Key: ************************
22:18:51 [TRACE] (1) e621_downloader::program: [src\program.rs:72] Login Download Favorites: true
22:18:51 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:72] SenderClient initializing with USER_AGENT_VALUE "e621_downloader/1.7.1 (by McSib <mcsibsart@gmail.com> on e621)"
22:18:51 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:64] Prompt for safe mode...
22:18:54 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:77] Safe mode decision: true
22:18:54 [TRACE] (1) e621_downloader::program: [src\program.rs:79] Parsing tag file...
22:18:56 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:379] No alias was found for score:>50...
22:18:56 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:380] Printing trace message for why None was returned...
22:18:56 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:381] error decoding response body: invalid type: map, expected a sequence at line 1 column 0
22:18:56 [TRACE] (1) e621_downloader::program: [src\program.rs:84] Parsing user blacklist...
22:18:57 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:103] Grabbing posts...
22:18:57 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 1 of tag fav:catuncle
22:18:58 [INFO] [38;5;39m[3m"fav:catuncle"[0m grabbed!
22:18:58 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 1 of tag krystal score:>50
22:19:01 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 2 of tag krystal score:>50
22:19:02 [DEBUG] (1) e621_downloader::e621::sender: Downloading page 3 of tag krystal score:>50
22:19:02 [INFO] [38;5;39m[3m"krystal score:>50"[0m grabbed!
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:244] Total file size for all images grabbed is 892974251KB
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "Single Posts"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        ""
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] Collection Post Length:     "0"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/Single Posts"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:217] Collection Single Posts is finished downloading...
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "fav:catuncle"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        ""
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] Collection Post Length:     "0"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/fav_catuncle"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:217] Collection fav:catuncle is finished downloading...
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:170] Printing Collection Info:
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:171] Collection Name:            "krystal score:>50"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:172] Collection Category:        "General Searches"
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:173] **Collection Post Length:     "377"**
22:19:02 [TRACE] (1) e621_downloader::e621: [src\e621\mod.rs:177] Static file path for this collection: "downloads/General Searches\krystal score__50"

Is it because the program can only capture up to 1800 files?
(Sorry for the question again lol))

**Comment by McSib** [comment]
Ah....

> [src\e621\sender\mod.rs:379] No alias was found for score:>50...
> 22:18:56 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:380] Printing trace message for why None was returned...
> 22:18:56 [TRACE] (1) e621_downloader::e621::sender: [src\e621\sender\mod.rs:381] error decoding response body: invalid type: map, expected a sequence at line 1 column 0

It appears you have a parser bug happening with the score tag. I'll have to look into that when I have time to. As for the 1,280 post limit on just the `Krystal` tag alone, read [here](https://github.com/McSib/e621_downloader#why-does-the-program-only-grab-only-1280-posts-with-certain-tags).

**Comment by McSib** [comment]
Could you send me your logfile so I may test the program using it? It will help me replicate your environment so I can get the same bug on my machine.

**Comment by McSib** [comment]
After doing some debugging, there were a couple of things I noticed from your log that pointed to the cause of the issue. What is happening isn’t a bug, the program is actually working as intended. The lines in the logfile specifying issues with aliases not being found is intended, though the wording did come off like a bug, so I will be correcting that. Onto the cause of your problems.
1.	With the first download with just the `Krystal` tag, I already went into detail and linked you to why there is a 1,280 download limit. I know it is frustrating to have a limit like that, but I must ensure I do not stress the server on bandwidth with large downloads (that and the occasional person trying to download way too much data like the entire `mammal` tag). There is a way around that, either by ordering by score (`order:score`), or by setting your date down so it will only grab posts after a certain date. I may modify this limit in the future to be more forgiving and a little larger for character tags, but so far, this is how the downloader is intended.
2.	For your search `Krystal score:>50`, you went into safe mode, this mode will download images and perform searches through e926, which will exclude any questionable or explicit work. I assessed this myself and got 377 posts downloaded, but when I cut off safe mode, I had the full 1,280 posts as expected. So, your problem was just an accidental switch into safe mode. You are prompted every time you start the program on whether you wish to enter safe mode, so make sure to answer no if you are trying to download explicit or questionable work under that tag.

Beyond that, this issue should be fixed now. However, if you end up having another issue or bug, feel free to post another issue on the board and I will look at it whenever I have the time to do so.
